Sources/Basic/FileSystem.swift

```diff
@@ -467,5 +466,60 @@ public class InMemoryFileSystem: FileSystem {
     }
 }
 
+/// A rerooted view on an existing FileSystem.
+///
+/// This is a simple wrapper which creates a new FileSystem view into a subtree
+/// of an existing filesystem. This is useful for passing to clients which only
+/// need access to a subtree of the filesystem but should otherwise remain
+/// oblivious to its concrete location.
+public struct RerootedFileSystemView: FileSystem {
```

Comment:
Comment is great, but perhaps also add some clarification that this doesn't provide any type of security, e.g. a symlink is allowed to escape the subtree.